### ISSUE TITLE
Possible problem with region lookup in param_mem_to_user_va()?

### ISSUE THREAD

**Issue by weo** [OP-TEE/optee_os#4060]
Could it be that commit 2667e1359e51 is causing many tests to fail on my system? The following patch solves all problems, no more errors are reported by xtest: 

```
------------------------- core/arch/arm/mm/tee_mmu.c --------------------------
index 94d6f75cb51e..cdeb91ee05d8 100644
@@ -843,7 +843,7 @@ static TEE_Result param_mem_to_user_va(struct user_mode_ctx *uctx,
 		phys_offs += mem->offs;
 		if (phys_offs < region->offset)
 			continue;
-		if (phys_offs >= (region->offset  - region->size))
+		if (phys_offs >= (region->offset + region->size))
 			continue;
 		va = region->va + phys_offs - region->offset;
 		*user_va = (void *)va;
```
I have only recently started working with OP-TEE, so unfortunately I can't judge whether the patch makes any sense.

**Comment by etienne-lms** [comment]
Your comment makes sense, there is no point subtracting the region size from its offset position.
This minus sign seems a typo to be fixed. I think @jenswi-linaro will confirm.
@weo, do you want to create a P-R to propose a fixup change?

**Comment by weo** [comment]
> @weo, do you want to create a P-R to propose a fixup change?
Okay

**Comment by etienne-lms** [comment]
Addressed by https://github.com/OP-TEE/optee_os/pull/4061.

**Comment by jenswi-linaro** [comment]
Makes sense.

**Comment by etienne-lms** [comment]
Thanks @weo for the finding and fix.

**Comment by etienne-lms** [comment]
can close this issue. addressed by #4061.
